### ISSUE TITLE
Automatic IP address get repeated IP address

### ISSUE THREAD

**Issue by Kwan9** [ntop/n2n#929]
Hi,  when edge draw IP addresses from the super node(skip -a ), the ip address is often repeated, how can i fix this problem ? 

Example:

Edge A 
Mac address : 00-FF-D6-7D-AF-9E
IPv4: 172.17.9.45
OS: WIndows 7/10

Edge B
Mac address :00-FF-C9-09-DB-0A
IPv4: 172.17.9.45
OS: WIndows 7/10

![image](https://user-images.githubusercontent.com/10485340/148989799-b9675f67-ac5a-4e5e-b098-d44b5eff3ad7.png)

**Comment by Logan007** [comment]
The IP address should not conflict if you only use only one supernode.

Conflicts could occur if you use federated supernodes.

To solve it, please know that the assigned IP address strongly depends on the hostname or edge's `-I ...` assigned identifier – if provided. So, in this case, just try `-I test` (or any other identifier looking good to you) at one of the edges.

**Comment by Kwan9** [comment]
I use only one super node ， ip address always conflict⋯

**Comment by Logan007** [comment]
Did `-I ...` with different identifiers help?

**Comment by Kwan9** [comment]
I‘m testing ’-I‘，will give more infoermation later ， i need to run more vmware first.  Thx

**Comment by Kwan9** [comment]
Tested , 5 edge nodes  using  '-I' with different identifiers , the problem  can resolve . Will try more nodes.

**Comment by ysn2382617094wsw** [comment]
Is there a detailed solution? Using -I on the edge has no effect, the two edges will still have the same IP

**Comment by Logan007** [comment]
`-I <identifier>`with different identifiers such as `-I ABC` and `-I DEF` should give different IP addresses.

**Comment by ysn2382617094wsw** [comment]
My supernode: ./supernode_upx.exe -p port -M -v
my edge1: ./edge_upx.exe -c test1 -l server:port -I Admin
my edge2: ./edge_upx.exe -c test1 -l server:port -I Users
Am I doing it right?
I am using version 3.1.1

**Comment by Logan007** [comment]
Yes, it looks good and should draw different IP addresses from the supernode.

One more thought  as I see that you use `-M` at the supernode. Do the VPN drivers at the edges have different MAC addresses, i.e. could you try some different `-m a0:a0:00:00:0:01` and `-m a0:a0:00:00:00:02` at the edges?

**Comment by ysn2382617094wsw** [comment]
Okay, I'll go and try, thank you very much for your advice, without your help I really don't know what to do anymore

**Comment by Logan007** [comment]
And if still not working, you could still try to manually assign IP addresses like `-a 10.10.10.1` and `-a 10.10.10.2` to the edges.

**Comment by ysn2382617094wsw** [comment]
Well, N2N's auto-acquisition IP has serious problems and can't be used in general, I know-a but I think I'd be happy if supernodes could do the work of automatically assigning IP, which would make things very simple

**Comment by ywnsya** [comment]
> 好吧，N2N的自动获取IP存在严重问题，通常无法使用，我知道，但我认为如果超级节点可以自动分配IP的工作，我会很高兴，这将使事情变得非常简单。

Why not try to use DHCP? It may works : )
